### PR TITLE
feat: [CDS-83530]:logs-async-steps

### PR DESCRIPTION
**Pull request by litvinovstas**
### Summary
1.Provided stream logs for async steps.
Test cases:
1.Executed pipeline with async K8s step, logs streaming working correctly

#### Screenshots

NA

#### PR Checklist

<details>
<summary>Please check if your PR fulfils the following requirements</summary>

- [ ] Tests for the changes have been added. Ideally, include a test that fails without this PR but passes with it.
- [ ] Docs have been [added/updated](https://harness.atlassian.net/jira/software/c/projects/DOC/boards/40).
</details>

<details>
<summary>Use the following comments to re-trigger PR Checks</summary>

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Sonar: `retrigger sonar`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Feature Name Check: `trigger featurenamecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress Rest: `retrigger cypress-rest`
- Cypress CD: `retrigger cypress-cd`
- Cypress Pipeline: `retrigger cypress-pipeline`
- Cypress CV: `retrigger cypress-cv`
- Fix Prettier: `fix prettier`
- MessageMetadata: `trigger messagecheck`
- GitLeaks: `trigger gitLeaks`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

### Rebase Instructions:
To rebase your branch onto the latest changes in the target branch, simply use the following command in a comment: `/rebase`

### CODE REVIEW

File: src/modules/70-pipeline/components/LogsContent/LogsState/createSections.ts [hunk 1]
```diff
@@ -5,7 +5,7 @@
  * https://polyformproject.org/wp-content/uploads/2020/06/PolyForm-Shield-1.0.0.txt.
  */
 
-import { defaultTo, findLast, set, snakeCase } from 'lodash-es'
+import { defaultTo, findLast, set, snakeCase, isEmpty } from 'lodash-es'
```
Comment (samarth-sharma31) [inline src/modules/70-pipeline/components/LogsContent/LogsState/createSections.ts:8]: Please attach video recording of the behaviour in the PR

File: src/modules/70-pipeline/components/LogsContent/LogsState/createSections.ts [hunk 2]
```diff
@@ -76,16 +76,28 @@ export function createSections(state: State, action: Action<ActionType.CreateSec
...
-      /* istanbul ignore else */
-      if (row.unitName) {
-        progressMap.set(row.unitName, {
-          status: row.status as UnitLoadingStatus,
-          startTime: row.startTime,
-          endTime: row.endTime
-        })
-      }
-    })
+    if (isEmpty(node.unitProgresses) && !isEmpty(node?.progressData?.unitProgresses)) {
```
Comment (samarth-sharma31) [inline src/modules/70-pipeline/components/LogsContent/LogsState/createSections.ts:79]: Also can we add jest coverage for both the if and else blocks

File: src/modules/70-pipeline/components/LogsContent/LogsState/createSections.ts [hunk 3]
```diff
@@ -76,16 +76,28 @@ export function createSections(state: State, action: Action<ActionType.CreateSec
...
+          progressMap.set(row.unitName, {
+            status: row.status as UnitLoadingStatus,
+            startTime: row.startTime,
+            endTime: row.endTime
+          })
+        }
+      })
+    } else {
+      node.unitProgresses.forEach(row => {
+        /* istanbul ignore else */
```
Comment (samarth-sharma31) [inline src/modules/70-pipeline/components/LogsContent/LogsState/createSections.ts:91]: lets remove this by adding coverage for this block

File: src/modules/70-pipeline/components/LogsContent/LogsState/createSections.ts [hunk 4]
```diff
@@ -76,16 +76,27 @@ export function createSections(state: State, action: Action<ActionType.CreateSec
...
+      node.unitProgresses.forEach(row => {
+        if (row.unitName) {
+          progressMap.set(row.unitName, {
+            status: row.status as UnitLoadingStatus,
+            startTime: row.startTime,
+            endTime: row.endTime
+          })
+        }
+      })
+    }
```
[the remark this answers is not in the document]
Comment (ishantmehtaharness) [reply src/modules/70-pipeline/components/LogsContent/LogsState/createSections.ts:99]: A better approach would be to get it in similar structure from Backend

**Review by chetanvekariya-harness — APPROVED** [review]

**Pull request closed without merging**